app/models/miq_server/worker_management/monitor.rb

```diff
@@ -63,6 +62,25 @@ def sync_workers
     result
   end
 
+  def cleanup_failed_workers
+    check_not_responding
+    check_pending_stop
+    clean_worker_records
+
+    if podified?
+    elsif systemd?
+      cleanup_failed_systemd_services
```

Comment:
Maybe when we get podified committed, we can make worker monitor subclasses where one type gets instantiated earlier, and these subclasses implements `cleanup_failed_things`, etc. so we don't need to have the `podified?` and `system?` checks.  For now, this is super clean and fixes just the issue we care about without complicating things prematurely.

Reply:
Yeah absolutely, I think with proper subclassing we can do the common cleanup in the base class and the specific cleanups in the subclass with `super` to do the core stuff